lib/elixir/lib/calendar.ex

```diff
@@ -194,6 +234,16 @@ defmodule Date do
   end
 
   @doc """
+  Returns the number of days in a week.
+
+  Always returns 7 days in a week irrespective of the
+  calendar in use.
+  """
+  def days_in_a_week() do
```

Comment:
I meant to suggest `{RD, microseconds}`. We are having a discussion on the forum about this though.

> For calculations on {RD, microseconds} (or my original case of integer) like the kday functions, there is no longer the concept of a calendar so an assumption has to be made.

Sorry, I am confused. All of those functions receives the calendar. Couldn't we just ask the calendar how many days there are in a week before doing the conversion to/from integer?

It seems that the most important computation being done in the `kday` functions is to find how many days until the beginning of next week. So what if we introduce those computations and still make them calendar sensitive? For example:

    Date.days_until_beginning_of_week(date) + 2 - 1
What if we generalize the `kday` functions to `beginning_of_the_week` and `beginning_of_next_week`. Those will be calendar dependent function that receives a date and return a date that is the beginning of the same week (or next week). This way, we don't need to perform the rounding calculations and the `days_in_week` function should no longer be necessary. Developers can still achieve the "next monday" feature